### PR TITLE
Prachi branch save event as text file

### PR DESCRIPTION
**Pull request by prachi2023**
Maybe @brandonywl you can check if I have saved all the details necessary for the event file to be loaded up again properly In case I missed a variable.

### CODE REVIEW

**Review by Chongjx — COMMENTED** [review]
Looks ok to me, just have to confirm that all necessary info are saved so that can use ParseAddEventCommand for loading.

**Review by Chongjx — APPROVED** [review]

**Review by R-Ramana — APPROVED** [review]

File: src/main/java/seedu/notus/storage/StorageManager.java [hunk 1]
```diff
@@ -280,4 +310,62 @@ public boolean noteExists(Note note, boolean isArchive) {
...
+            eventDetails = getEventDetailsSaveFormat(event);
+            eventDetails += PrefixSyntax.PREFIX_DELIMITER + PrefixSyntax.PREFIX_RECURRING
+                        + " " + event.getRecurrenceType() + " "
+                        + PrefixSyntax.PREFIX_DELIMITER + PrefixSyntax.PREFIX_STOP_RECURRING
+                        + " " + event.getEndRecurrenceDate() + " "
+                        + LS;
+            fwAppend.write(eventDetails);
+        }
+        fwAppend.close();
+    }
```
Comment (brandonywl) [inline src/main/java/seedu/notus/storage/StorageManager.java:352]: So it's basically to convert the event back to it's input line and stored right? E.g. add-e /t Tutorial /timing 2020-10-29 13:00 /repeat weekly /reminder 1-day. Looks great!

File: src/main/java/seedu/notus/storage/StorageManager.java [hunk 2]
```diff
@@ -122,6 +145,30 @@ public void loadAllNotes(Notebook notebook, Timetable timetable, TagManager tagM
...
+        } catch (FileNotFoundException exception) {
+            throw new SystemException(SystemException.ExceptionType.EXCEPTION_FILE_NOT_FOUND_ERROR);
+        }
+        while (s.hasNext()) {
+            String eventDetails = AddEventCommand.COMMAND_WORD + " " +  s.nextLine();
+            Command command = parserManager.parseCommand(eventDetails);
+            command.setData(notebook, timetable, tagManager, this);
+            command.execute();
+        }
+        s.close();
```
Comment (brandonywl) [inline src/main/java/seedu/notus/storage/StorageManager.java:169]: I feel this is a great way to save and load!

File: src/main/java/seedu/notus/data/timetable/Event.java [hunk 3]
```diff
@@ -167,6 +170,26 @@ public boolean getRecurring() {
...
+            }
+        }
+
+        if (weekRepeatPeriod != null) {
+            for (Integer unit : weekRepeatPeriod) {
+                periods.add(unit + "-week");
+            }
+        }
+        return periods;
+    }
```
Comment (brandonywl) [inline src/main/java/seedu/notus/data/timetable/Event.java:191]: Was implementing this on my branch but since it's here it's great!

**Review by brandonywl — APPROVED** [review]
Looks good to me!

File: src/main/java/seedu/notus/storage/StorageManager.java [hunk 4]
```diff
@@ -280,4 +310,62 @@ public boolean noteExists(Note note, boolean isArchive) {
...
+    private static String getEventDetailsSaveFormat(Event event) {
+        String eventDetails;
+        eventDetails = PrefixSyntax.PREFIX_DELIMITER + PrefixSyntax.PREFIX_TITLE + " " + event.getTitle() + " "
+                + PrefixSyntax.PREFIX_DELIMITER + PrefixSyntax.PREFIX_TIMING + " " + event.getDateTime() + " ";
+
+        ArrayList<String> reminderPeriods = event.getReminderPeriodsString();
+
+        for (String reminderPeriod : reminderPeriods) {
+            eventDetails += PrefixSyntax.PREFIX_DELIMITER + PrefixSyntax.PREFIX_REMIND + " " + reminderPeriod + " ";
+        }
```
Comment (brandonywl) [inline src/main/java/seedu/notus/storage/StorageManager.java:363]: Actually for reminder, it accepts the inputs as /remind 1-day 2-day 3-day. Not just /remind 1-day /remind 2-day

**Review by brandonywl — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/notus/storage/StorageManager.java (hunk 4 above)
Comment (prachi2023) [reply src/main/java/seedu/notus/storage/StorageManager.java:363]: Ah alright ok will fix that thanks!

Comment (brandonywl) [reply src/main/java/seedu/notus/storage/StorageManager.java:363]:
```suggestion
        eventDetails += (reminderPeriods.size() > 1) ? PrefixSyntax.PREFIX_DELIMITER + PrefixSyntax.PREFIX_REMIND  : "";
        for (String reminderPeriod : reminderPeriods) {
            eventDetails +=  " " + reminderPeriod;
        }
```

File: src/main/java/seedu/notus/data/timetable/Event.java [hunk 5]
```diff
@@ -167,6 +170,26 @@ public boolean getRecurring() {
         return isRecurring;
     }
 
+    public ArrayList<String> getReminderPeriodsString() {
+        ArrayList<String> periods = new ArrayList<>();
+
+        ArrayList<Integer> dayRepeatPeriod = reminderPeriods.get(REMINDER_DAY);
+        ArrayList<Integer> weekRepeatPeriod = reminderPeriods.get(REMINDER_WEEK);
```
Comment (brandonywl) [inline src/main/java/seedu/notus/data/timetable/Event.java:177]: Actually, what would happen if reminderPeriods that is stored was null? Should have a checker here if not a empty arraylist will be given

Comment (prachi2023) [reply src/main/java/seedu/notus/data/timetable/Event.java:177]: Yup Ive added a check for it

**Pull request merged**